test/mock_osystem.cpp

```diff
@@ -0,0 +1,132 @@
+#define FORBIDDEN_SYMBOL_ALLOW_ALL
+
+#include "mock_osystem.h"
+#include "backends/fs/posix/posix-fs-factory.h"
+
+namespace Common {
+
+class OSystem_Mock : public OSystem {
+public:
+	OSystem_Mock() {
+		_fsFactory = new POSIXFilesystemFactory();
+	}
+	void logMessage(LogMessageType::Type type, const char *message) {
```

Comment:
Can you please mark all those functions as override? Some compilers are quite noisy when compiling with c++11 support and omitting this identifier.